django/template/base.py

```diff
@@ -852,6 +852,13 @@ def __init__(self, var):
                         "Variables and attributes may "
                         "not begin with underscores: '%s'" % var
                     )
+                # Disallow characters that are allowed in numbers but not in a
+                # variable name.
+                for c in ["+", "-"]:
+                    if c in var:
+                        raise TemplateSyntaxError(
+                            "Invalid character ('%s') in variable name: '%s'" % (c, var)
+                        )
```

Comment:
I was going to avoid commenting on this but then I saw you inlined the var Sarah 👍… you could go one step further in readability: 🤔 
```suggestion
                if "+" in var or "-" in var:
                    raise TemplateSyntaxError(
                        "Invalid character ('%s') in variable name: '%s'" % (c, var)
                    )
```
which also happens to be marginally quicker though I doubt it makes much difference here 🤔 (something which I'm grateful for Adam to point out to me)

Reply:
Oh the other thing I was going to comment on: Was it not possible to solve this in the same area of code as the related ticket?

The PR description wasn't clear that solving this upstream wasn't viable so just wanted to confirm that?  Usually these sorts of bolt-on afterthoughts are what leads to spaghetti 🍝 🙂

Reply:
The suggestion doesn't work as c is not defined
> The PR description wasn't clear that solving this upstream wasn't viable so just wanted to confirm that? 

I had given this a go but wasn't able to, do you have a suggestion?

Reply:
> I had given this a go but wasn't able to, do you have a suggestion?

Nope all good 👍

Reply:
True, but this is also the same for variables and attributes that begin with an underscore